File: Server/Server/Server/Model/DTO/GameStateDTO.cs
namespace Server.Model.DTO
{
    /// <summary>
    /// Représente l'état d'une partie à un moment donnée pour le transfert au client
    /// </summary>
    public class GameStateDTO
    {
        private string board;

        private int capturedBlack;

        private int capturedWhite;

        private int moveNumber;

        /// <summary>
        /// Renvoi l'état du board au moment de cet état de la partie
        /// </summary>
        public string Board { get => board; }

        /// <summary>
        /// Renvoi le nombre de pierres capturées par les noirs à ce moment de la partie
        /// </summary>
        public int CapturedBlack { get => capturedBlack; }

        /// <summary>
        /// Renvoie le nombre de pierres capturées par les blancs à ce moment de la partie
        /// </summary>
        public int CapturedWhite { get => capturedWhite; }

        /// <summary>
        /// Renvoi le numéro du coup de cet état de la partie
        /// </summary>
        public int MoveNumber { get => moveNumber; }

        public GameStateDTO(string board, int capturedBlack, int capturedWhite, int moveNumber)
        {
            this.board = board;
            this.capturedBlack = capturedBlack;
            this.capturedWhite = capturedWhite;
            this.moveNumber = moveNumber;
        }
    }
}

File: Client/src/app/Model/DTO/GameStateDTO.ts
export class GameStateDTO {
    private board: string;
    private capturedBlack: number;
    private capturedWhite: number;
    private moveNumber: number;

    public constructor(board: string, capturedBlack: number, capturedWhite: number, moveNumber: number)
    {
        this.board = board;
        this.capturedBlack = capturedBlack;
        this.capturedWhite = capturedWhite;
        this.moveNumber = moveNumber;
    }

    public Board():string{
        return this.board;
    }
    public CapturedBlack():number{
        return this.capturedBlack;
    }
    public CapturedWhite():number{
        return this.capturedWhite;
    }
    public MoveNumber():number{
        return this.moveNumber;
    }
}